packages/bot-skeleton/src/scratch/blocks/Logic/controls_if.js

```diff
@@ -71,160 +72,203 @@ Blockly.Blocks.controls_if = {
-                return;
-            }
+        this.rendered = false;
 
-            const old_mutation_dom = this.mutationToDom();
-            const new_input_num = this.else_if_count + 1;
+        if (updateFn) {
+            updateFn.call(this);
+        }
 
-            if (this.else_count === 0) {
-                // No `elseif`, just add an `else`-statement
-                this.appendDummyInput('ELSE_LABEL').appendField(localize('else'));
-                this.appendDummyInput('DELETE_ELSE').appendField(this.getRemoveInputIcon(new_input_num, true));
-                this.appendStatementInput('ELSE');
+        this.updateShape();
+        this.rendered = is_rendered;
+        this.initSvg();
 
-                this.else_count++;
-            } else {
-                // We've already got else + elseifs, keep adding elseifs into infinity.
-                this.appendDummyInput(`IF_LABEL${new_input_num}`).appendField(localize('else if'));
-                this.appendValueInput(`IF${new_input_num}`).setCheck('Boolean');
-                this.appendDummyInput(`THEN_LABEL${new_input_num}`).appendField(localize('then'));
-                this.appendDummyInput(`DELETE_ICON${new_input_num}`).appendField(
-                    this.getRemoveInputIcon(new_input_num, false)
-                );
-                this.appendStatementInput(`DO${new_input_num}`);
-
-                this.else_if_count++;
-            }
+        const group = Blockly.Events.getGroup();
+        const new_mutation_dom = this.mutationToDom();
+        const new_mutation = new_mutation_dom && Blockly.Xml.domToText(new_mutation_dom);
 
-            // We already have an else, this input needs to be moved to the bottom where it belongs.
-            if (this.getInput('ELSE')) {
-                this.moveInputBefore('ELSE_LABEL', null);
-                this.moveInputBefore('DELETE_ELSE', null);
-                this.moveInputBefore('ELSE', null);
-            }
+        if (old_mutation !== new_mutation) {
+            const change_event = new Blockly.Events.BlockChange(this, 'mutation', null, old_mutation, new_mutation);
+            Blockly.Events.fire(change_event);
 
-            // Move plus-icon to the bottom
-            this.moveInputBefore('MUTATOR', null);
+            setTimeout(() => {
+                Blockly.Events.setGroup(group);
+                this.bumpNeighbours_();
+                Blockly.Events.setGroup(false);
+            }, Blockly.BUMP_DELAY);
+        }
 
-            this.initSvg();
+        if (this.rendered) {
             this.render();
+        }
 
-            // Fire a mutation event so this can be undone/redone.
-            const old_mutation = Blockly.Xml.domToText(old_mutation_dom);
-            const new_mutation = Blockly.Xml.domToText(this.mutationToDom());
+        Blockly.Events.setGroup(false);
+    },
+    updateShape() {
+        if (this.getInput('ELSE')) {
+            this.removeInput('ELSE');
```

Comment:
Does this remove the conditional block and its children blocks?

Reply:
It will remove all the blocks that were attached to the `ELSE`'s statement input, however we do store these connections so that in the of the user clicking undo or redo we can restore these connections.